src/main/java/com/google/api/codegen/discovery/transformer/java/JavaSampleMethodToViewTransformer.java

```diff
@@ -85,19 +85,29 @@ private SampleView createSampleView(SampleTransformerContext context) {
         typeTable.getRequestTypeName(config.apiTypeName(), methodInfo.requestType()).getNickname();
 
     List<SampleFieldView> fields = new ArrayList<>();
+    List<SampleFieldView> optionalFields = new ArrayList<>();
     List<String> fieldVarNames = new ArrayList<>();
     for (FieldInfo field : methodInfo.fields().values()) {
-      SampleFieldView sampleFieldView = createSampleFieldView(field, typeTable, symbolTable);
+      SampleFieldView sampleFieldView = createSampleFieldView(field, context, symbolTable);
       fields.add(sampleFieldView);
-      fieldVarNames.add(sampleFieldView.name());
+      if (sampleFieldView.required()) {
+        fieldVarNames.add(sampleFieldView.name());
+      } else {
+        optionalFields.add(sampleFieldView);
+      }
```

Comment:
Probably the right move here is to fix the names,

`fields` -> `requiredFields`
`fieldVarNames` -> `apiCallArgs`
`optionalFields` -> `optionalFields`

does that seem reasonable?